magento/magento2
# Allow customer to specify associated product qtys when adding grouped product to cart via RESTful API

### Description (*)

This PR addresses the issue described here: [magento/magento2#26909](https://github.com/magento/magento2/issues/26909)
Currently, when adding a grouped product to cart via RESTful API, there is no way to way to specify the quantities for the individual associated products. Additionally, there is no supporting documentation for adding a grouped product to the cart in the [docs](https://devdocs.magento.com/guides/v2.3/rest/tutorials/orders/order-add-items.html)

### Related Pull Requests


### Fixed Issues (if relevant)

1. magento/magento2#26909: Add grouped product with individual quantity to cart using rest api

### Manual testing scenarios (*)


These steps assume Magento sample data is installed. Otherwise, update the body of the calls to match your database.

1. Generate a customer token by sending a post request to the **rest/V1/integration/customer/token** endpoint with a body of:
`{
	"username": "roni_cost@example.com",
	"password": "roni_cost3@example.com"
}`

2. Create an empty cart by sending a post request to the **rest/V1/carts/mine** endpoint. The bearer token should be the token returned by step 1 and the body should be empty.

3. Add a grouped product to cart by sending a post request to the **rest/V1/carts/mine/items** endpoint with a body of:
**The quote_id parameter in the body should be the quote ID returned by step 2. The bearer token should be the token returned by step 1.**

`{
    "cartItem": { 
        "sku": "gr", 
        "qty": 1,
        "quote_id": "5",
        "product_option": { 
            "extension_attributes": 
            {
                "grouped_options": [
                    { "id": 2, "qty": 3}, { "id": 3, "qty": 2}
                ]
            } 
        } 
    }
}`

4. Get the cart contents by sending a get request to the **rest/V1/carts/mine** endpoint. The bearer token should be the token returned by step 1 and the body should be empty. 

### Questions or comments


### Contribution checklist (*)
